client/extensions/woocommerce/app/products/product-form-categories-card.js

```diff
@@ -14,24 +14,27 @@ import FormFieldSet from 'components/forms/form-fieldset';
 import FormLabel from 'components/forms/form-label';
 import TokenField from 'components/token-field';
 import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import { generateProductCategoryId } from 'woocommerce/state/ui/product-categories/actions';
 
 // TODO Rename this card since it contains other controls, and may contain more in the future (like tax)
 const ProductFormCategoriesCard = (
-	{ siteId, product, productCategories, editProduct, translate }
+	{ siteId, product, productCategories, editProduct, editProductCategory, translate }
 ) => {
 	const handleChange = ( categoryNames ) => {
 		const newCategories = compact( categoryNames.map( ( name ) => {
 			const category = find( productCategories, { name: escape( name ) } );
 
 			if ( ! category ) {
```

Comment:
Not for now, I wonder if we are doing too much in our presentation layer and not enough in the UI state.